sklearn/inspection/_partial_dependence.py

```diff
@@ -172,6 +179,95 @@ def _partial_dependence_brute(est, grid, features, X, response_method):
     return averaged_predictions
 
 
+def _validate_parameters(estimator, X, features, response_method, method):
+    if not (is_classifier(estimator) or is_regressor(estimator)):
+        raise ValueError(
+            "'estimator' must be a fitted regressor or classifier."
+        )
+
+    if isinstance(estimator, Pipeline):
+        # TODO: to be removed if/when pipeline get a `steps_` attributes
+        # assuming Pipeline is the only estimator that does not store a new
+        # attribute
+        for est in estimator:
+            # FIXME: remove the None option when it will be deprecated
+            if est not in (None, 'drop'):
+                check_is_fitted(est)
+    else:
+        check_is_fitted(estimator)
+
+    if (is_classifier(estimator) and
+            isinstance(estimator.classes_[0], np.ndarray)):
+        raise ValueError(
+            'Multiclass-multioutput estimators are not supported'
+        )
+
+    # Use check_array only on lists and other non-array-likes / sparse. Do not
+    # convert DataFrame into a NumPy array.
+    if not(hasattr(X, '__array__') or sparse.issparse(X)):
+        X = check_array(X, force_all_finite='allow-nan', dtype=np.object)
+
+    accepted_responses = ('auto', 'predict_proba', 'decision_function')
+    if response_method not in accepted_responses:
+        raise ValueError(
+            'response_method {} is invalid. Accepted response_method names '
+            'are {}.'.format(response_method, ', '.join(accepted_responses)))
+
+    if is_regressor(estimator) and response_method != 'auto':
+        raise ValueError(
+            "The response_method parameter is ignored for regressors and "
+            "must be 'auto'."
+        )
+
+    accepted_methods = ('brute', 'recursion', 'auto')
+    if method not in accepted_methods:
+        raise ValueError(
+            'method {} is invalid. Accepted method names are {}.'.format(
+                method, ', '.join(accepted_methods)))
+
+    if method == 'auto':
+        if (isinstance(estimator, BaseGradientBoosting) and
+                estimator.init is None):
+            method = 'recursion'
+        elif isinstance(estimator, BaseHistGradientBoosting):
+            method = 'recursion'
+        else:
+            method = 'brute'
+
+    if method == 'recursion':
+        if not isinstance(estimator,
+                          (BaseGradientBoosting, BaseHistGradientBoosting)):
+            supported_classes_recursion = (
+                'GradientBoostingClassifier',
+                'GradientBoostingRegressor',
+                'HistGradientBoostingClassifier',
+                'HistGradientBoostingRegressor',
+            )
+            raise ValueError(
+                "Only the following estimators support the 'recursion' "
```

Comment:
Use an f-string

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/2894ebe79543504a5cbfe06382dc3ff6effa17a8